relay-event-schema/src/protocol/nel.rs

```diff
@@ -0,0 +1,249 @@
+        Request {
+            url: self.url.clone(),
+            headers,
+            ..Request::default()
+        }
+    }
+    pub fn apply_to_event(data: &[u8], event: &mut Event) -> Result<(), NelError> {
+        let mut nel: Annotated<Nel> =
+            Annotated::from_json_bytes(data).map_err(NelError::InvalidJson)?;
+
+        if let Some(nel) = nel.value_mut() {
+            if let Some(body) = nel.body.value() {
+                event.logentry = Annotated::new(LogEntry::from({
+                    if nel.ty.value().map_or("", |v| v.as_str()) == "http.error" {
+                        format!(
+                            "{} / {} ({})",
+                            body.phase.as_str().unwrap_or(""),
+                            body.ty.as_str().unwrap_or(""),
+                            body.status_code.value().unwrap_or(&0)
+                        )
+                    } else {
+                        format!(
+                            "{} / {}",
+                            body.phase.as_str().unwrap_or(""),
+                            body.ty.as_str().unwrap_or(""),
+                        )
+                    }
+                }));
+
+                event.request = Annotated::new(nel.get_request());
+                event.logger = Annotated::from("nel".to_string());
+
+                // Exrtact common tags.
+                let tags = event.tags.get_or_insert_with(Tags::default);
+                if let Some(ref method) = body.method.value() {
+                    tags.push(Annotated::new(TagEntry::from_pair((
+                        Annotated::new("method".to_string()),
+                        Annotated::new(method.to_string()),
+                    ))));
+                }
+                if let Some(ref protocol) = body.protocol.value() {
+                    tags.push(Annotated::new(TagEntry::from_pair((
+                        Annotated::new("protocol".to_string()),
+                        Annotated::new(protocol.to_string()),
+                    ))));
+                }
+                if let Some(ref status_code) = body.status_code.value() {
+                    tags.push(Annotated::new(TagEntry::from_pair((
+                        Annotated::new("status_code".to_string()),
+                        Annotated::new(status_code.to_string()),
+                    ))));
+                }
+                if let Some(ref server_ip) = body.server_ip.value() {
+                    tags.push(Annotated::new(TagEntry::from_pair((
+                        Annotated::new("server_ip".to_string()),
+                        Annotated::new(server_ip.to_string()),
+                    ))));
+                }
+            }
+
+            // Set the timestamp on the event when it actually occured.
+            let now: DateTime<Utc> = Utc::now();
+            let event_time = now.sub(Duration::milliseconds(*nel.age.value().unwrap_or(&0)));
+            event.timestamp = Annotated::new(Timestamp::from(event_time));
```

Comment:
`.sub()` might panic.

```suggestion
            if let Some(event_time) = now.checked_sub_signed(Duration::milliseconds(*nel.age.value().unwrap_or(&0))) {
                event.timestamp = Annotated::new(Timestamp::from(event_time));
            }
```